doc/pages/Manual.md

```diff
@@ -7,33 +7,228 @@ primarily of use for packagers, package maintainers and repository maintainers.
+- by setting the `OPAMSWITCH=<switch>` environment variable, to set it within a
+  single shell session. This can be done by running `eval $(opam config env
+  --switch <switch>)` to set the shell environment at once, see below.
+
+Switches have their own prefix, normally `~/.opam/<switch>`, where packages get
+intalled ; to use what is installed in a switch, some environment variables need
+to be set, _e.g._ to make executables installed into `~/.opam/<switch>/bin`
+visible, that directory needs to be added to `PATH`, but individual packages can
+define their own settings as well.
+
+Command `opam config env` returns the environment updates corresponding to the
+current switch, in a format readable by your shell, and when needed opam will
+prompt you to run:
+
+```
+eval $(opam config env)
+```
 
-## General file format
+A switch is created using `opam switch create <switch> (<compiler>|--empty)`.
+- `<switch>` can be either a plain name, or a directory name (if containing `/`
+  or starting with `.`). In the latter case the switch is _local_ and instead of
+  being held at `~/.opam/<switch>`, it will be created in the given directory,
+  as a `_opam` subdirectory. Local switches are automatically selected depending
+  on the current directory, see above.
+- If a `<compiler>` is selected, opam will install the corresponding packages
+  and their dependencies in the new switch. These packages will be marked as
+  _base_, protected against removal and unaffected by upgrade commands.
+  `<compiler>` can be selected among packages which have the `compiler` flag
+  set, or their versions. Use `opam switch list-available` to list them.
+
+#### Structure
+
+If we define `<switch-prefix>` as:
+- `~/.opam/<switch>` for plain switches
+- `<switch>/_opam` for local switches, when `<switch>` is a path
+
+Switches are laid out thusly:
+- `<switch-prefix>/`: prefix of the switch, holding the installation hierarchy
+  in the UNIX `/usr` standard (with subdirectories `bin`, `lib`, `share`, `man`,
+  `doc`, `etc`...)
+- `<switch-prefix>/.opam-switch/`: holds all opam data regarding this switch
+- [`<switch-prefix>/.opam-switch/switch-config`](#switchconfig): switch-specific
+  configuration
+- [`<switch-prefix>/.opam-switch/switch-state`](#switchstate): stores the sets
+  of installed, base, pinned packages
+- `<switch-prefix>/.opam-switch/environment`: contains the environment variable
+  settings for this switch
+- `<switch-prefix>/.opam-switch/reinstall`: list of packages marked for
+  reinstallation (development packages where changes have been detected)
+- [`<switch-prefix>/.opam-switch/config/<pkgname>.config`](#ltpkgnamegtconfig):
+  installed package's, opam specific configuration
+- [`<switch-prefix>/.opam-switch/install/<pkgname>.install`](#ltpkgnamegtinstall):
+  `.install` files used to install the given package
+- `<switch-prefix>/.opam-switch/install/<pkgname>.changes`: file changes done by
```

Comment:
yes, this basically records an abridged diff corresponding to the package installation